solidity/contracts/StakingPortBacker.sol

```diff
@@ -0,0 +1,273 @@
+/**
+▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+  ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓▌        ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+  ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+  ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+
+                           Trust math, not hardware.
+*/
+
+pragma solidity 0.5.17;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/SafeERC20.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+
+import "./KeepToken.sol";
+import "./TokenGrant.sol";
+import "./StakeDelegatable.sol";
+import "./TokenStaking.sol";
+import "./libraries/RolesLookup.sol";
+import "./utils/BytesLib.sol";
+import "./TokenSender.sol";
+
+/// @title StakingPortBacker
+/// @notice Provides additional liquidity from the primary token supply for
+/// token owners staking on the previous staking contract version and letting
+/// them to stake on the new staking contract version without having to wait
+/// for the entire undelegation period to unlock their tokens.
+///
+/// It lets the Keep team to make an amount of KEEP tokens available to
+/// temporarily stake on behalf of a token owner, provided that they have those
+/// same tokens currently locked up in the old staking contract. Undelegation of
+/// all tokens in the new staking contract would be blocked until the
+/// temporarily-staked tokens provided by the Keep team are repaid by the token
+/// owner.
+///
+/// The expected mode of usage for users who have already staked on the old
+/// contract is to:
+/// 1. Copy stake.
+/// 2. Undelegate from the old staking contract.
+/// 3. Operate as normal on the new staking contract.
+/// 4. Once the old staking contract's undelegated balance is available for
+/// recovery, recover it and use the recovered tokens to repay the
+/// StakingPortBacker contract.
+contract StakingPortBacker is Ownable {
```

Comment:
Do we want to include this contract in `2_deploy_contracts.js`?

Reply:
I... don't think so. We don't have two staking contracts there.

Reply:
Perhaps based on an environment variable or similar? I don't think it needs to block this PR though.

Reply:
`StakingPortBacker` [is implemented with an assumption](https://github.com/keep-network/keep-core/pull/1906#discussion_r454402750) that the "old" staking contract version does not support top-ups and does not work with escrow. If we wanted to simulate mainnet situation, we should deploy the previous version of `TokenStaking` as well. I am not sure how important it is, we'll probably need it only one time when testing the migration.